### PR TITLE
Void-Creation

### PR DESCRIPTION
**Pull request by The-Marvex**

### CODE REVIEW

File: org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
+package org.bridgedb.tools.qc;
```
Comment (egonw) [inline org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java]: Here you need to add a header, showing who wrote it and what the license is. See other BridgeDb code for examples.

File: org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import org.bridgedb.bio.DataSourceTxt;
+import org.bridgedb.rdb.SimpleGdb;
+import org.bridgedb.rdb.SimpleGdbFactory;
+
+import java.io.*;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class VoIDTool {
```
Comment (egonw) [inline org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java]: Please add JavaDocs.

File: org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    private PrintStream out;
+    private final File oldDb;
+
+    private SimpleGdb oldGdb;
+    public FileWriter file ;
+
+    String fileName = "rdf.txt";
+    public VoIDTool(File f1) throws IDMapperException, IOException {
```
Comment (egonw) [inline org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java]: Please add JavaDoc.

File: org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        this(f1, System.out);
+        file = new FileWriter("rdf.txt", true);
+    }
+    public VoIDTool(File f1, OutputStream out) throws IOException {
```
Comment (egonw) [inline org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java]: Please add JavaDoc.

File: org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        oldDb= f1;
+        this.out = new PrintStream(out);
+    }
+    public void appendStrToFile(String fileName,
```
Comment (egonw) [inline org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java]: Please add JavaDoc.

File: void.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+#
+# Run after first compiling with "mvn compile install" and updating the version
+# number below.
+
+VERSION=`tail -1 org.bridgedb/version.props | cut -d'=' -f2`
+CLASSPATH=org.bridgedb.tools.qc/target/org.bridgedb.tools.qc-${VERSION}.jar:\
```
Comment (egonw) [inline void.sh]: Please check the CLASSPATH. Does it really need the QC tool?

File: org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    }
+    public void createVoid(String db1)throws IOException{
+        db1 = db1.toLowerCase();
+        switch(db1) {
+            case "gmpl":
+                appendStrToFile(fileName,"<http://rdf.wikipathways.org/20190713/rdf/gpml> \n");
+                appendStrToFile(fileName,"        a                 dcat:Distribution ;\n");
+                appendStrToFile(fileName,"void:dataDump     <https://jenkins.bigcat.unimaas.nl/job/GPML%20to%20GPML%20+%20WP%20RDF/ws/WP2RDF/output/gpml/*zip*/gpml.zip> ;\n");
+                appendStrToFile(fileName,"dcat:downloadURL  <https://jenkins.bigcat.unimaas.nl/job/GPML%20to%20GPML%20+%20WP%20RDF/ws/WP2RDF/output/gpml/*zip*/gpml.zip> ;\n");
+                appendStrToFile(fileName,"\n");
```
Comment (egonw) [inline org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java]: I do not think hardcoding data sources is the way forward. We want this tool to work on any arbitrary .bridge (Derby) file. Please replace the `switch` approach with code that retrieves all the information from:

1. whatever is in the Derby file
2. whatever you can learn from the `datasources.ttl`

Point 1 here links to how to store metadata in Derby files. That may need extension and I welcome your input.

Comment (The-Marvex) [reply org.bridgedb.tools.qc/src/org/bridgedb/tools/void/VoIDTool.java]: we need to store the data in the derby file while creating them, so the first thing I think I would do is change the create-bridgedb-metabolites to store the data in the derby file. Next would be to modify the VoIDTool to access the stored data in the derby file. 
I'll go through the datasources.ttl to see what I can salvage from it.

**Comment by egonw** [comment]
@The-Marvex, did you update this branch with changes? I am not sure if this PR is now ready for inclusion.

**Comment by AlasdairGray** [comment]
I agree with Egon that we should not be hardcoding the approach. 
(I'm too far removed from the code at the moment to be able to get it up and running quickly, but would be happy to go through the approach if it needs further discussion.)

**Pull request closed without merging**